Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+      }
+    }
+    else // cancel previous selection and select a new row
+    {
+      if (selectedRows != undefined && selectedRows.Length > 0) //TODO what to do here if selectedRows is undefined? should we throw an error before we even get here?
+      {
+        //  Unselect all rows and clear the list
+        BocList.UnselectAllRows (bocList, isRowHighlightingEnabled);
+      }
+      //  Add currentRow to list and select it
+      BocList.SelectRow (bocList, currentRowBlock, isRowHighlightingEnabled);
+    }
+    try
+    {
+      selectorControl.focus();
+    }
+    catch (e)
+    {
+    }  
+    BocList._isRowSelectorClickExecuting = false;
+    return true;
+  }
+
+  //  Selects a row.
+  //  Adds the row to the _bocList_selectedRows array and increments _bocList_selectedRowsLength.
+  //  bocList: The BocList to which the row belongs.
+  //  rowBlock: The row to be selected.
+  //  isRowHighlightingEnabled: true to mark the selected rows via css-class
+  public static SelectRow(bocList: HTMLElement, rowBlock: BocList_RowBlock, isRowHighlightingEnabled: boolean): void
+  {
+    //  Add currentRow to list  
+    var selectedRows = BocList._selectedRows[bocList.id];
+    if (selectedRows != undefined) //TODO what to do here? most of the calls rely on selectedRows being defined
+    {
+      selectedRows.Rows[rowBlock.SelectorControl.id] = rowBlock;
+      selectedRows.Length++;
+    }
+    // Select currentRow
+    rowBlock.SelectorControl.checked = true;
+    if (isRowHighlightingEnabled)
+      $(rowBlock.Row).addClass(BocList.TrClassNameSelected);
+    
+    if (selectedRows != undefined)
+    {
+      BocList.SetSelectAllRowsSelectorOnDemand (selectedRows);
+    }
+  }
+
+  //  Unselects all rows in a BocList.
+  //  Clears _bocList_selectedRows array and sets _bocList_selectedRowsLength to zero.
+  //  bocList: The BocList whose rows should be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  public static UnselectAllRows(bocList: HTMLElement, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id];
+    for (var rowID in selectedRows?.Rows)
+    {
+      var rowBlock = selectedRows?.Rows[rowID];
+      if (rowBlock != null)
+      {
+        BocList.UnselectRow (bocList, rowBlock, isRowHighlightingEnabled);
+      }
+    }
+    
+    //  Start over with a new array
+    selectedRows?.Clear();
+  }
+
+  //  Unselects a row.
+  //  Removes the row frin the _bocList_selectedRows array and decrements _bocList_selectedRowsLength.
+  //  bocList: The BocList to which the row belongs.
+  //  rowBlock: The row to be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  public static UnselectRow(bocList: HTMLElement, rowBlock: BocList_RowBlock, isRowHighlightingEnabled: boolean): void
+  {
+    //  Remove currentRow from list
+    var selectedRows = BocList._selectedRows[bocList.id];
```

Comment:
Sames as above. Assert non-nullability using `!` and restore the old code without the ifs.